chroma_core/lib/util.py

```diff
@@ -225,16 +225,64 @@ def normalize_nid(string):
     return string
 
 
+IML_ACTION_RUNNER_SOCKET_PATH = "http+unix://%2Fvar%2Frun%2Fiml-action-runner.sock/"
+
+
 class RustAgentCancellation(Exception):
     pass
 
 
-def invoke_rust_agent(host, command, args={}, cancel_event=Event()):
+def invoke_rust_local_action(command, args={}, cancel_event=Event()):
     """
     Talks to the iml-action-runner service
     """
 
-    SOCKET_PATH = "http+unix://%2Fvar%2Frun%2Fiml-action-runner.sock/"
+    request_id = uuid.uuid4()
+
+    trigger = Event()
+
+    class ActionResult:
+        ok = None
+        error = None
+
+    def start_action(ActionResult, trigger):
+        try:
+            ActionResult.ok = requests_unixsocket.post(
+                IML_ACTION_RUNNER_SOCKET_PATH,
+                json={"LOCAL": {"type": "ACTION_START", "action": command, "args": args, "id": str(request_id)}},
+            ).content
+        except Exception as e:
+            ActionResult.error = e
+        finally:
+            trigger.set()
+
+    t = Thread(target=start_action, args=(ActionResult, trigger))
+
+    t.start()
+
+    # Wait for action completion, waking up every second to
+    # check cancel_event
+    while True:
+        if cancel_event.is_set():
+            requests_unixsocket.post(
```

Comment:
This will need to be updated to work with docker.

Reply:
Yes, but your patch hasn't landed yet.

Whichever patch lands first, the other will need to rebase and update..